paddle/fluid/operators/tensorrt/tensorrt_engine_op.h

```diff
@@ -80,19 +90,95 @@ class TensorRTEngineOp : public framework::OperatorBase {
     input_names_ = Inputs("Xs");
     max_batch_size_ = Attr<int>("max_batch_size");
     workspace_size_ = Attr<int>("workspace_size");
+    enable_int8_ = Attr<bool>("enable_int8");
+    calibration_data_ = Attr<std::string>("calibration_data");
+    engine_key_ = Attr<std::string>("engine_key");
 
     auto params = Attr<std::vector<std::string>>("parameters");
     for (const auto &param : params) {
       param_names_.insert(param);
     }
+    // calibration_mode is ture represents we need to
+    // generate the calibration table data.
+    calibration_mode_ = (enable_int8_ && calibration_data_.size() == 0);
+
+    VLOG(4) << "calibration_mode: " << calibration_mode_;
+    if (enable_int8_ && calibration_data_.size()) {
+      calibrator_.reset(new TRTInt8Calibrator(calibration_data_));
+    }
   }
 
  protected:
+  void RunNativeImpl(const framework::Scope &scope,
+                     const platform::Place &dev_place) const {
+    framework::Executor executor(dev_place);
+    auto *block = Attr<framework::BlockDesc *>("sub_block");
+    auto *program = block->Program();
+    auto *scope_ptr = const_cast<framework::Scope *>(&scope);
+    auto ctx = executor.Prepare(*program, block->ID());
+    executor.RunPreparedContext(ctx.get(), scope_ptr, false, true, true);
+  }
+
   void RunImpl(const framework::Scope &scope,
                const platform::Place &dev_place) const override {
+    if (calibration_mode_ == true) {
+      RunCalibration(scope, dev_place);
+      return;
+    }
     RunTrt(scope, dev_place);
   }
 
+  void RunCalibration(const framework::Scope &scope,
+                      const platform::Place &dev_place) const {
+    // This process will builds a 32-bit trt engine, runs it on the calibration
+    // set, and records a histogram for each
+    // tensor of the distribution of activation values.
+    LOG(INFO) << "Running calibration trt int8 ...";
```

Comment:
The log
```
82788 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:58.889819 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:58.996390 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:59.103065 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:59.209903 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:59.320547 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:59.428273 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:59.535076 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
I0116 03:40:59.641471 10328 tensorrt_engine_op.h:132] Running calibration trt int8 ...
```

is duplicated, use `LOG_FIRST_N`, or add some unique information to each log, for example, 

```
LOG(INFO) << "Running calibration trt int8 thread " << std::this_thread::get_id();
```

Reply:
Done!